solutions/bronze/cf-581D.mdx

```diff
@@ -0,0 +1,83 @@
+---
+id: cf-581D
+source: CF
+title: Three Logos
+author: Chongtian Ma
+---
+
+[Official Editorial](https://codeforces.com/blog/entry/20508)
+
+## Implementation
+
+We can brute force all rotations of the three rectangles with a bitmask of $2^3$, with bit $i$ on representing the $i$'th rectangle should be rotated by
+$90$ degrees. The rest is just checking if the configurations are valid.
+
+**Time Complexity:** $\mathcal{O}(2^n \cdot \textbf{side}^4)$, where $n$ is the number of rectangles (in this case, 3) and $\textbf{side}$ as the side length of the square.
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+int main() {
+	int n = 3;
+	vector<pair<int, int>> logos(n);
+	for (int i = 0; i < n; i++) {
+		scanf("%d%d", &logos[i].first, &logos[i].second);
+	}
+	long long area = 0;
+	for (pair<int, int> p : logos) { area += p.first * p.second; }
+	// if area is not a perfect square, then it's already invalid
+	int len = 1;
+	while (len * len < area) { len++; }
+	if (len * len != area) {
+		printf("-1");
+		return 0;
+	}
+	// loop through all rotations of each rectangle
+	for (int rotate_mask = 0; rotate_mask < (1 << n); rotate_mask++) {
+		vector<string> grid(len, string(len, 'Z'));  // temporary grid
+		// 'Z' represents an empty space in the grid
+		int num_placed = 0;
+		for (int i = 0; i < len; i++) {
+			for (int j = 0; j < len; j++) {
+				if (grid[i][j] == 'Z') {
```

Comment:
space this thing out jesus christ

Reply:
huh?

Reply:
space relevant code blocks out
my eyes are burning